src/bot/integrations/tipeeestream.ts

```diff
@@ -0,0 +1,182 @@
+import _ from 'lodash';
+import chalk from 'chalk';
+import axios from 'axios';
+import io from 'socket.io-client';
+
+import Integration from './_interface';
+import { onChange, onStartup } from '../decorators/on.js';
+import { settings } from '../decorators';
+import { ui } from '../decorators.js';
+import { error, info, tip } from '../helpers/log.js';
+import { triggerInterfaceOnTip } from '../helpers/interface/triggers.js';
+
+import { getRepository } from 'typeorm';
+import { User, UserTipInterface } from '../database/entity/user';
+import api from '../api.js';
+import events from '../events.js';
+import users from '../users.js';
+import eventlist from '../overlays/eventlist.js';
+import currency from '../currency';
+import alerts from '../registries/alerts.js';
+
+class TipeeeStream extends Integration {
+  socketToTipeeestream: SocketIOClient.Socket | null = null;
+
+  @settings()
+  @ui({ type: 'text-input', secret: false })
+  username = '';
+
+  @settings()
+  @ui({ type: 'text-input', secret: true })
+  apiKey = '';
+
+  constructor () {
+    super();
+  }
+
+  @onStartup()
+  @onChange('enabled')
+  onStateChange (key: string, val: boolean) {
+    if (val) {
+      this.connect();
+    } else {
+      this.disconnect();
+    }
+  }
+
+  async disconnect () {
+    if (this.socketToTipeeestream !== null) {
+      this.socketToTipeeestream.removeAllListeners();
+      this.socketToTipeeestream.disconnect();
+      this.socketToTipeeestream = null;
+    }
+  }
+
+  @onChange('apiKey')
+  async connect () {
+    this.disconnect();
+
+    if (this.username.trim() === '' || this.apiKey.trim() === '' || !this.enabled) {
+      return;
+    }
+
+    // get current avaliable host and port
+    // destucted response from https://api.tipeeestream.com/v2.0/site/socket
+    // example response: { "code": 200, "message": "success", "datas": { "port": "443", "host": "https://sso-cf.tipeeestream.com" } }
+    const { data: { datas: { host, port }}} = await axios.get('https://api.tipeeestream.com/v2.0/site/socket');
+
+    this.socketToTipeeestream = io(`${host}:${port}`,
```

Comment:
try
```suggestion
    this.socketToTipeeestream = io.connect(`${host}:${port}`,
```

sole `io` is trying to reuse connections, which may cause doubling

Reply:
Got same behavior even with .connect()